silver_payu/payment_processors.py

```diff
@@ -114,30 +112,58 @@ def execute_transaction(self, transaction):
     def _charge_transaction(self, transaction):
         token = transaction.payment_method.token
 
-        billing_details = transaction.payment_method.archived_customer
+        customer_details = transaction.payment_method.archived_customer
         try:
             delivery_details = {
-                "DELIVERY_ADDRESS": billing_details["BILL_ADDRESS"],
-                "DELIVERY_CITY": billing_details["BILL_CITY"],
-                "DELIVERY_EMAIL": billing_details["BILL_EMAIL"],
-                "DELIVERY_FNAME": billing_details["BILL_FNAME"],
-                "DELIVERY_LNAME": billing_details["BILL_LNAME"],
-                "DELIVERY_PHONE": billing_details["BILL_PHONE"]
+                "DELIVERY_ADDRESS": customer_details["BILL_ADDRESS"],
+                "DELIVERY_CITY": customer_details["BILL_CITY"],
+                "DELIVERY_EMAIL": customer_details["BILL_EMAIL"],
+                "DELIVERY_FNAME": customer_details["BILL_FNAME"],
+                "DELIVERY_LNAME": customer_details["BILL_LNAME"],
+                "DELIVERY_PHONE": customer_details["BILL_PHONE"]
             }
         except KeyError as error:
             transaction.fail(fail_reason=text_type('Invalid customer details. [{}]').format(error))
             transaction.save()
             return False
 
         payment_details = {
-            "AMOUNT": str(transaction.amount),
-            "CURRENCY": str(transaction.currency),
-            "EXTERNAL_REF": str(transaction.uuid)
+            "PRICES_CURRENCY": str(transaction.currency),
+            "ORDER_REF": str(transaction.uuid),
+            "PAY_METHOD": "CCVISAMC",
         }
-        payment_details.update(billing_details)
+
+        if transaction.document:
+            pname = text_type("{provider} {doc_type} {doc_number}").format(
+                doc_type=transaction.document.kind,
+                doc_number=transaction.document.series_number,
+                provider=transaction.provider.name
+            )
+
+            vat = str(int(transaction.document.sales_tax_percent))
```

Comment:
Maybe we'll have some problems with vat with decimals.

Reply:
VAT with decimals are not supported by PayU (confirmed by their support), and I'm not aware of such values in real use cases.